src/core/lib/surface/legacy_channel.cc

```diff
@@ -97,9 +97,23 @@ absl::StatusOr<OrphanablePtr<Channel>> LegacyChannel::Create(
                                 .value_or(CoreConfiguration::Get()
                                               .resolver_registry()
                                               .GetDefaultAuthority(target));
+    experimental::StatsPluginChannelScope scope(target, authority);
     *(*r)->stats_plugin_group =
-        GlobalStatsPluginRegistry::GetStatsPluginsForChannel(
-            experimental::StatsPluginChannelScope(target, authority));
+        GlobalStatsPluginRegistry::GetStatsPluginsForChannel(scope);
+    // Add per-channel stats plugins.
+    auto* stats_plugin_list = args.GetPointer<
+        std::shared_ptr<std::vector<std::shared_ptr<StatsPlugin>>>>(
+        GRPC_ARG_STATS_PLUGINS);
+    if (stats_plugin_list != nullptr) {
+      for (const auto& plugin : **stats_plugin_list) {
+        bool is_enabled = false;
+        std::shared_ptr<StatsPlugin::ScopeConfig> config;
+        std::tie(is_enabled, config) = plugin->IsEnabledForChannel(scope);
+        if (is_enabled) {
+          (*r)->stats_plugin_group->AddStatsPlugin(plugin, std::move(config));
```

Comment:
@yashykt, didn't we say that we would need to store a bit in the `StatsPluginGroup` indicating whether each plugin has come from the global registry or was directly added to the channel, since we'll need that information to properly propagate stats plugins to child channels?

Reply:
We shouldn't need to, since we already have the channel arg that only has the per channel stats plugins.